server/src/internalClusterTest/java/org/elasticsearch/snapshots/GetSnapshotsIT.java

```diff
@@ -208,6 +209,34 @@ public void testPaginationRequiresVerboseListing() throws Exception {
         );
     }
 
+    public void testFilterBySLMPolicy() throws Exception {
+        final String repoName = "test-repo";
+        createRepository(repoName, "fs");
+        createNSnapshots(repoName, randomIntBetween(1, 5));
+        final List<SnapshotInfo> snapshotsWithoutPolicy = clusterAdmin().prepareGetSnapshots("*").setSnapshots("*").get().getSnapshots();
+        final String snapshotWithPolicy = "snapshot-with-policy";
+        final String policyName = "some-policy";
+        final SnapshotInfo withPolicy = assertSuccessful(
+            clusterAdmin().prepareCreateSnapshot(repoName, snapshotWithPolicy)
+                .setUserMetadata(Map.of(SnapshotsService.POLICY_ID_METADATA_FIELD, policyName))
+                .setWaitForCompletion(true)
+                .execute()
+        );
+
+        final List<SnapshotInfo> snapshotsWithPolicy = clusterAdmin().prepareGetSnapshots("*")
+            .setSnapshots("*")
+            .setPolicies(policyName)
+            .get()
+            .getSnapshots();
+        assertThat(snapshotsWithPolicy, is(List.of(withPolicy)));
+        final List<SnapshotInfo> snapshotsNoPolicy = clusterAdmin().prepareGetSnapshots("*")
+            .setSnapshots("*")
+            .setPolicies("*", "-" + policyName)
+            .get()
+            .getSnapshots();
+        assertThat(snapshotsNoPolicy, is(snapshotsWithoutPolicy));
```

Comment:
Can we add tests of:

1. A slm-policy that is not there.
2. Multiple slm-policies (where some may not exist).
3. An actual inclusion pattern, i.e., `policyName + "*"`.
4. An actual exclusion pattern, i.e., `*, "-" + policyName + "*"

Could be randomly picking from some of these if that is easiest.

Reply:
++ Added all 4 cases explicitly to rest and internal cluster tests :)